megaparsec-tests/tests/Text/Megaparsec/Byte/MachineIntsSpec.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Text.Megaparsec.Byte.MachineIntsSpec (spec) where
+
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import Data.Void
+import Data.Word
+import Test.Hspec
+import Test.Hspec.Megaparsec
+--import Test.QuickCheck
+import Text.Megaparsec
+import Text.Megaparsec.Byte.MachineInts
+
+-- TODO: use QuickCheck
+
+type Parser = Parsec Void ByteString
+
+spec :: Spec
+spec = do
+  describe "anyLE" $ do
+    it "parses a little-endian Word32" $ do
+      let p = anyLE @Word32 <* eof
+          (s, _, valueLE) = exW32
+      prs p s `shouldParse` valueLE
+      prs' p s `succeedsLeaving` ""
+
+----------------------------------------------------------------------------
+-- Helpers
+
+prs ::
```

Comment:
Why do you re-define these functions here? They are available in `Test.Hspec.Megaparsec.AdHoc`.

Reply:
Oops, copy-pasted wrong. Thanks.

Reply:
Wait, no, `Test.Hspec.Megaparsec.AdHoc` defines a parser over a `String` stream. `ByteSpec` and `Byte.LexerSpec` both redefine it for `ByteString` instead, which I need too. What should I do here? Leave at copy-paste? ByteString Parser module? Stream type typevar in `Parser`?